### PR TITLE
Change sapm to expect effective_irradiance in W/m2

### LINKED ISSUE #472: Effective irradiance units
`pvsystem.sapm` expected `effective_irradiance` in units of suns (1 sun = 1000 W/m2).  `pvsystem.calcparams_desoto` expects `effective_irradiance` in W/m2.  I want to change to consistent units for effective irradiance.  

Preferences for suns or W/m2?

### PR DESCRIPTION
**Pull request by cwhanse**
pvlib python pull request guidelines
====================================

Thank you for your contribution to pvlib python! You may delete all of these instructions except for the list below.

You may submit a pull request with your code at any stage of completion.

The following items must be addressed before the code can be merged. Please don't hesitate to ask for help if you're unsure of how to accomplish any of the items below:

 - [x] Closes #472
 - [x] I am familiar with the [contributing guidelines](http://pvlib-python.readthedocs.io/en/latest/contributing.html).
 - [x] Fully tested. Added and/or modified tests to ensure correct behavior for all reasonable inputs. Tests (usually) must pass on the TravisCI and Appveyor testing services.
 - [ ] Updates entries to `docs/sphinx/source/api.rst` for API changes.
 - [ ] Adds description and name entries in the appropriate `docs/sphinx/source/whatsnew` file for all changes.
 - [x] Code quality and style is sufficient. Passes LGTM and SticklerCI checks.
 - [x] New code is fully documented. Includes sphinx/numpydoc compliant docstrings and comments in the code where necessary.
 - [ ] Pull request is nearly complete and ready for detailed review.

Brief description of the problem and proposed solution (if not already fully described in the issue linked to above):

### CODE REVIEW

File: pvlib/pvsystem.py [hunk 1]
```diff
@@ -2010,21 +2000,16 @@ def sapm_effective_irradiance(poa_direct, poa_diffuse, airmass_absolute, aoi,
...
+        The SAPM effective irradiance in W/m^2.
     """
 
     F1 = sapm_spectral_loss(airmass_absolute, module)
     F2 = sapm_aoi_loss(aoi, module)
 
-    E0 = reference_irradiance
-
-    Ee = F1 * (poa_direct*F2 + module['FD']*poa_diffuse) / E0
+    Ee = F1 * (poa_direct*F2 + module['FD']*poa_diffuse)
```
[the remark this answers is not in the document]
Comment (wholmgren) [reply pvlib/pvsystem.py]: this can be ignored. you can even add E226 to the skip list in the stickler config file

**Comment by wholmgren** [comment]
This feels like a change for 0.7 instead of 0.6.1 because it will break code. I'm not opposed to moving forward with a 0.7 release and skipping a 0.6.1 release.

**Comment by cwhanse** [comment]
OK with me if we leave this for 0.7.

**Comment by cwhanse** [comment]
OK, leaving this PR without `whatsnew` changes for 0.7

**Comment by wholmgren** [comment]
@cwhanse do you no longer want to change the units? If so, we should close the corresponding issue. If not, perhaps we should merge this and make 0.7 the next release. I also have no objections to leaving it open during a 0.6.2 cycle.

**Comment by cwhanse** [comment]
I still want to change the units.  I was doing some housecleaning in my fork and deleted this branch without realizing that it is connected to this PR.

**Pull request closed without merging**